### PR TITLE
fix: fix to correct type in walle.ts

### PR DESCRIPTION
**Pull request by odanado**
The `wallet.ts` type deviates from Ethereum's JSON RPC definition.

### CODE REVIEW

File: src/wallet.ts [hunk 1]
```diff
@@ -9,59 +9,59 @@ import * as sigUtil from 'eth-sig-util';
...
 }
 
-interface WalletMiddlewareOptions {
+export interface WalletMiddlewareOptions {
   getAccounts: (req: JsonRpcRequest<unknown>) => Promise<string[]>;
   processDecryptMessage?: (
     msgParams: MessageParams,
     req: JsonRpcRequest<unknown>,
-  ) => Promise<Record<string, unknown>>;
+  ) => Promise<string>;
```
Comment (odanado) [inline src/wallet.ts:29]: `Returns` is a string type
https://docs.metamask.io/guide/rpc-api.html#eth-decrypt

File: src/wallet.ts [hunk 2]
```diff
@@ -9,59 +9,59 @@ import * as sigUtil from 'eth-sig-util';
...
   processEncryptionPublicKey?: (
     address: string,
     req: JsonRpcRequest<unknown>,
-  ) => Promise<Record<string, unknown>>;
+  ) => Promise<string>;
```
Comment (odanado) [inline src/wallet.ts:33]: https://docs.metamask.io/guide/rpc-api.html#eth-getencryptionpublickey

File: src/wallet.ts [hunk 3]
```diff
@@ -9,59 +9,59 @@ import * as sigUtil from 'eth-sig-util';
...
   processEthSignMessage?: (
     msgParams: MessageParams,
     req: JsonRpcRequest<unknown>,
-  ) => Promise<Record<string, unknown>>;
+  ) => Promise<string>;
```
Comment (odanado) [inline src/wallet.ts:37]: https://eth.wiki/json-rpc/API

File: src/wallet.ts [hunk 4]
```diff
@@ -9,59 +9,59 @@ import * as sigUtil from 'eth-sig-util';
...
     txParams: TransactionParams,
     req: JsonRpcRequest<unknown>,
-  ) => Promise<Record<string, unknown>>;
+  ) => Promise<string>;
   processTypedMessage?: (
     msgParams: MessageParams,
     req: JsonRpcRequest<unknown>,
     version: string,
-  ) => Promise<Record<string, unknown>>;
+  ) => Promise<string>;
```
Comment (odanado) [inline src/wallet.ts:54]: https://eips.ethereum.org/EIPS/eip-712

**Comment by odanado** [comment]
@mcmire @Gudahtt
Can I review this pull request?

**Review by Gudahtt — APPROVED** [review]
LGTM! Sorry for the delay in reviewing this.

I reviewed each of these type changes, and confirmed that in practice the types we are passing match what you've done here.

**Pull request merged**